protos/hub/hub.proto

```diff
@@ -25,6 +25,35 @@ service HubQueryService {
   rpc GetSettings(SettingsReq) returns (SettingsResp);
   rpc GetHosts(HostsReq) returns (HostsResp);
   rpc GetStats(StatsReq) returns (StatsResp);
+  rpc GetNodesLocation(NodesReq) returns (NodesLocationsResp);
+}
+
+message NodesReq {
+  repeated string node_id = 1;
+  string requester_id = 2;
+  bytes signature = 3;
+}
+
+message NodeLocation {
+  string node_id = 1;
+  string region = 2;
+  string country = 3;
+  string version = 4;
+  float latitude = 5;
+  float longitude = 6;
+  string host_address = 7;
+  string name = 8;
+  string error_msg = 9;
+}
+
+message NodesLocationsResp {
+  repeated Host nodes = 1;
+  string preparer_id = 2;
```

Comment:
what is preparer_id here?

Reply:
Should be hub's id to identify hub and signature , to avoid some cheater to provide information. Not sure hub have it now. Lets talk .